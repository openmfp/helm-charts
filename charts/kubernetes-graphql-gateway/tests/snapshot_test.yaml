suite: Test the Template files
values:
- ../values.yaml
release:
  name: openmfp
  namespace: openmfp-system
chart:
  appVersion: 1.0.0
tests:
- it: match the Snapshot
  asserts:
  - matchSnapshot: {}
