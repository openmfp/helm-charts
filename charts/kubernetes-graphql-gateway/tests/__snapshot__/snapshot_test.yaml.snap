match the Snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: openmfp-kubernetes-graphql-gateway
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: openmfp-kubernetes-graphql-gateway
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: openmfp-kubernetes-graphql-gateway
            version: 1.0.0
        spec:
          containers:
            - args:
                - gateway
              env:
                - name: OPEN_API_DEFINITIONS_PATH
                  value: /app/definitions
                - name: LOCAL_DEVELOPMENT
                  value: "false"
                - name: ENABLE_KCP
                  value: "true"
                - name: GATEWAY_HANDLER_CFG_GRAPHI_QL
                  value: "true"
                - name: GATEWAY_LOG_LEVEL
                  value: trace
                - name: GATEWAY_PORT
                  value: "8080"
                - name: GATEWAY_USERNAMECLAIM
                  value: email
                - name: GATEWAY_SHOULDIMPERSONATE
                  value: "true"
              image: ghcr.io/openmfp/kubernetes-graphql-gateway:1.0.0
              imagePullPolicy: IfNotPresent
              name: openmfp-kubernetes-graphql-gateway-gateway
              ports:
                - containerPort: 8081
                  name: metrics
                  protocol: TCP
                - containerPort: 3389
                  name: health-port
                  protocol: TCP
                - containerPort: 8080
                  name: service-port
                  protocol: TCP
              resources:
                limits:
                  cpu: 200m
                  memory: 512Mi
                requests:
                  cpu: 100m
                  memory: 320Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /app/definitions
                  name: definitions-volume
            - args:
                - listener
              env:
                - name: OPEN_API_DEFINITIONS_PATH
                  value: /app/definitions
                - name: LOCAL_DEVELOPMENT
                  value: "false"
                - name: ENABLE_KCP
                  value: "true"
                - name: LISTENER_API_EXPORT_NAME
                  value: kcp.io
              image: ghcr.io/openmfp/kubernetes-graphql-gateway:1.0.0
              imagePullPolicy: IfNotPresent
              name: openmfp-kubernetes-graphql-gateway-listener
              ports:
                - containerPort: 8091
                  name: metrics
                  protocol: TCP
                - containerPort: 3390
                  name: health-port
                  protocol: TCP
                - containerPort: 8090
                  name: service-port
                  protocol: TCP
              resources:
                limits:
                  cpu: 200m
                  memory: 512Mi
                requests:
                  cpu: 100m
                  memory: 320Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /app/definitions
                  name: definitions-volume
          imagePullSecrets:
            - name: github
          serviceAccountName: openmfp-kubernetes-graphql-gateway
          volumes:
            - emptyDir:
                sizeLimit: 500Mi
              name: definitions-volume
  2: |
    apiVersion: security.istio.io/v1
    kind: AuthorizationPolicy
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      action: ALLOW
      rules:
        - to:
            - operation:
                methods:
                  - GET
                paths:
                  - '*/graphql'
        - from:
            - source:
                requestPrincipals:
                  - '*'
          to:
            - operation:
                methods:
                  - POST
                paths:
                  - '*/graphql'
      selector:
        matchLabels:
          app: openmfp-kubernetes-graphql-gateway
  3: |
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      host: openmfp-kubernetes-graphql-gateway.openmfp-system.svc.cluster.local
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
  4: |
    apiVersion: security.istio.io/v1beta1
    kind: PeerAuthentication
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      mtls:
        mode: STRICT
      selector:
        matchLabels:
          app: openmfp-kubernetes-graphql-gateway
  5: |
    apiVersion: security.istio.io/v1beta1
    kind: RequestAuthentication
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      jwtRules: null
      selector:
        matchLabels:
          app: openmfp-kubernetes-graphql-gateway
  6: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      gateways:
        - openmfp-system/gateway
      hosts:
        - '*'
      http:
        - corsPolicy:
            allowHeaders:
              - Authorization
              - Content-Type
              - '*'
            allowMethods:
              - GET
              - POST
            allowOrigins:
              - regex: .*
          match:
            - uri:
                prefix: /kubernetes-graphql-gateway/
          rewrite:
            uri: /
          route:
            - destination:
                host: openmfp-kubernetes-graphql-gateway.openmfp-system.svc.cluster.local
                port:
                  number: 8080
  7: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
  8: |
    apiVersion: v1
    kind: Service
    metadata:
      name: openmfp-kubernetes-graphql-gateway
      namespace: openmfp-system
    spec:
      ports:
        - name: http-8080
          port: 8080
          protocol: TCP
        - name: health
          port: 3389
          protocol: TCP
      selector:
        app: openmfp-kubernetes-graphql-gateway
      type: ClusterIP
